### PR TITLE
fix filter logic and support app without DOKKU_SCALE files in repo

### PR DESCRIPTION
**Pull request by michaelshobbs**

### CODE REVIEW

File: pre-deploy [hunk 1]
```diff
@@ -1,23 +1,46 @@
...
+    local COMMAND; local DOKKU_PROCTYPE_FILTER_REGEX
+    DOKKU_PROCTYPE_FILTER_REGEX="$(sed -r -e 's/([[:graph:]]+)[[:blank:]]?/^\1=|/g' <<< "$DOKKU_PROCTYPE_FILTER"| sed -e 's/|$//g')"
+    COMMAND="egrep \"$DOKKU_PROCTYPE_FILTER_REGEX\" \$SCALE_FILE > /tmp/DOKKU_SCALE.tmp ; "
+    COMMAND+="mv -f /tmp/DOKKU_SCALE.tmp \$SCALE_FILE"
+    if does_image_file_exist "$IMAGE" "$IMAGE_SCALE_FILE"; then
+      dokku_log_info1 "DOKKU_SCALE file found in container and DOKKU_PROCTYPE_FILTER defined"
+      dokku_log_info1 "scaling Procfile types -> $DOKKU_PROCTYPE_FILTER"
+      local cid;
+      cid=$(docker run -d -e SCALE_FILE="$IMAGE_SCALE_FILE" "$IMAGE" /bin/bash -c "$COMMAND")
+      test "$(docker wait "$cid")" -eq 0
```
Comment (u2mejc) [inline pre-deploy]: Double brackets for bash style guide goodness.

Comment (michaelshobbs) [inline pre-deploy]: This is directly copied from dokku

Comment (michaelshobbs) [inline pre-deploy]: ¯_(ツ)_/¯

Comment (u2mejc) [inline pre-deploy]: `[[ "$(docker wait "$cid")" -eq 0 ]]`

Comment (michaelshobbs) [inline pre-deploy]:
```
[[ "$(docker wait "$cid")" -eq 0 ]] && docker commit "$cid" "$IMAGE" > /dev/null
```

**Comment by michaelshobbs** [comment]
@u2mejc updated

**Pull request merged**